lib/permissions.js

```diff
@@ -7,16 +7,36 @@
 
 var vocab = require('./vocab')
 var rdf = require('./util/rdf-parser').rdflib
-var auth = require('./auth')
 var webClient = require('./web')
+var path = require('path')
+var md5 = require('md5')
+
+var inheritancePredicateName = 'defaultForNew'
 
 var PermissionType = {
-  Owner: 'Owner',
+  Control: 'Control',
   Read: 'Read',
   Write: 'Write',
   Append: 'Append'
 }
 
+// Represents an Agent object to distinguish singular agents vs user groups
+function Agent (agentURI, isSingular) {
+  this.uri = agentURI || ''
+  this.isSingular = typeof isSingular !== 'undefined' ? isSingular : true
+  this.isDefaultForNew = false
```

Comment:
Completely agree. Further refactoring, there will be Authorization object that encapsulates Agent, Permissions & resource URL. Next PR will have it.